src/network/components/StreamSearchDropdown.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { FunctionComponent, useEffect, useState } from 'react'
+import { useQuery } from '@tanstack/react-query'
+import { truncateStreamName } from '~/shared/utils/text'
+import { SearchDropdown } from '~/components/SearchDropdown'
+import {
+    checkIfStreamExists,
+    getPagedStreams,
+    TheGraphOrderBy,
+    TheGraphOrderDirection,
+    TheGraphStreamResult,
+} from '~/services/streams'
+
+export const StreamSearchDropdown: FunctionComponent<{
+    onStreamChange: (streamId: string) => void
+    name: string
+    streamId: string
+    disabled?: boolean
+}> = ({ onStreamChange, name, streamId, disabled }) => {
+    const [streamSearchValue, setStreamSearchValue] = useState('')
+
+    const streamsQuery = useQuery({
+        queryKey: ['createSponsorshipsStreamSearch', streamSearchValue],
+        queryFn: async (ctx) => {
+            const result: TheGraphStreamResult = await getPagedStreams(
+                20,
+                ctx.pageParam,
+                undefined,
+                streamSearchValue,
+                TheGraphOrderBy.Id,
+                TheGraphOrderDirection.Asc,
+            )
+
+            return result
+        },
+        staleTime: 60 * 1000, // 1 minute
+        keepPreviousData: true,
+    })
+
+    useEffect(() => {
+        if (!streamId || streamId === streamSearchValue) {
+            return
+        }
+        checkIfStreamExists(streamId).then((exists) => {
+            if (exists) {
+                onStreamChange(streamId)
+                setStreamSearchValue(streamId)
+            }
+        })
+    }, [streamId])
```

Comment:
This effect performs only when `streamId` changes (and on mount). Is that intentional? Can you resolve them eslint warnings (missing deps)?

Reply:
That's the intention - it's just to double-check if the streamId passed to the modal's .pop() function is valid. As for the warning - here I'll put an eselint-ignore comment because that's how it's intended to work and it works properly